### DISCUSSION TITLE
Deployment steps to serve nouns-webapp for production

### DISCUSSION THREAD

**Discussion by jackdishman** [nounsDAO/nouns-monorepo#685] [Frontend]
Hello! 

I'm working with the [Boston DAO](https://github.com/bostondao/nouns-monorepo-fork) to launch a fork of the `nouns-monorepo`. I'm looking to find steps to deploy `nouns-webapp` to production. I have initially tried the process using Vercel which failed to compile with the following errors:

```
../hidapi/linux/hid.c:44:10: fatal error: libudev.h: No such file or directory
...
make: *** [Release/obj.target/hidapi-linux-hidraw/hidapi/linux/hid.o] Error 1
19:49:02.234 | gyp ERR! build error
19:49:02.234 | gyp ERR! stack Error: `make` failed with exit code: 2
19:49:02.234 | gyp ERR! stack     at ChildProcess.onExit (/vercel/path0/node_modules/node-gyp/lib/build.js:194:23)
19:49:02.235 | gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
19:49:02.235 | gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
...
Error: Command "yarn run build" exited with 1
```

If anyone could walk me through the deployment process that Nouns DAO follows or point me in the direction of tooling used, that would be greatly appreciated!

Thanks in advance,
Jack

**Comment by mixtou** [comment]
What version of node are you using?

**Reply by mixtou** [reply]
It needs node 16. Use nvm to install node 16 and then nvm use 16 to change version. Then it compiles

**Reply by jackdishman** [reply]
Yeah I'm using node v16! It compiles locally fine, just having issues when deploying on a hosting service.

**Comment by spatializes** [comment]
@jackdishman did you figure this one out? Just hit the same error attempting to deploy to Vercel. Using Node 16.

**Reply by jackdishman** [reply]
Still haven't figured it out. Here's the issues I'm facing on our fork: https://github.com/bostondao/nouns-monorepo-fork/issues/28

**Comment by ghost** [comment]
Hey @jackdishman. I think the issue might be with the Vercel build environment. AFAICT, it runs Amazon Linux 2 and you'll need to install the `libudev-dev` library for some part of this build.

[These Vercel docs](https://vercel.com/docs/concepts/deployments/build-image#installing-something-else-in-the-build-image) should guide you on how to add a package to the Vercel build environment.

The package you'll want to install for Amazon Linux 2 is called `systemd-devel` (which should include `libudev`): https://amazonlinux.pkgs.org/2/amazonlinux-core-x86_64/systemd-devel-219-78.amzn2.0.16.x86_64.rpm.html

Hope this helps!
